### PR TITLE
feat: [PL-28646]: Add new Get File task in SCM to handle multiple operations as part of single delegate task

### PR DESCRIPTION
**Pull request by mohitg0795**
## Describe your changes

Added new delegate task to getFile from git. Tested it by integrating with Ng manager, but integration not added in this PR to avoid backward incompatibility.
Tried out with github and bitbucket flows, getFile is working fine. Error handling not included in this PR, will be included in consequent PR which will have integration with Ng manager.

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/38690)

### CODE REVIEW

File: 952-scm-java-client/src/main/java/io/harness/impl/ScmOrchestratorServiceImpl.java [hunk 1]
```diff
@@ -226,4 +228,9 @@ public GetLatestCommitOnFileResponse getLatestCommitOnFile(
       ScmConnector scmConnector, String branchName, String filepath) {
     return scmServiceGitClient.getLatestCommitOnFile(scmConnector, branchName, filepath);
   }
+
+  @Override
+  public GitFileResponse getFile(ScmConnector scmConnector, GitFileRequest gitFileContentRequest) {
+    return null;
```
Comment (adivishy1) [inline 952-scm-java-client/src/main/java/io/harness/impl/ScmOrchestratorServiceImpl.java:234]: was the null missed or is it handled on the caller side?

**Review by mohitg0795 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38690)* status: 41 file(s) to review. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`952-scm-java-client/src/main/java/io/harness/impl/ScmOrchestratorServiceImpl.java` line 234 at r9](https://reviewable.io/reviews/harness/harness-core/38690#-NE7oeOlwz082CDvePXA-r9-234:-NEFfF-Q850s7RS8C3mW:btnq56x) ([raw file](https://github.com/harness/harness-core/blob/36fd8a17a66b54243a8c365cb28070ad13fe07ff/952-scm-java-client/src/main/java/io/harness/impl/ScmOrchestratorServiceImpl.java#L234)):*
<details><summary><i>Previously, adivishy1 (Adithya Viswanathan) wrote…</i></summary><blockquote>

was the null missed or is it handled on the caller side?
</blockquote></details>

This must be a miss. As this code is yet to be tested, so its expected. Once E2E testing is done, such things won't be there.

**Review by adivishy1 — DISMISSED** [review]
> Reviewed 1 of 1 files at r3, 37 of 41 files at r8, 2 of 2 files at r9, 13 of 13 files at r10, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38690)* status: 42 file(s) to review. 1 of 2 approvals obtained. 

___
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/common/impl/ScmFacilitatorServiceImpl.java` line 239 at r10](https://reviewable.io/reviews/harness/harness-core/38690#-NEVxNWk2km_lkyY3D8g:-NEVxNWk2km_lkyY3D8h:b-fqea1v) ([raw file](https://github.com/harness/harness-core/blob/2ccabb31149525a02e6b21ca6775a662d1169dd8/136-git-sync-manager/src/main/java/io/harness/gitsync/common/impl/ScmFacilitatorServiceImpl.java#L239)):*
> ```Java
>                 GitFileRequest.builder()
>                     .branch(scmGetFileByBranchRequestDTO.getBranchName())
>                     .commitId(null)
> ```

even earlier without your code changes as well, the comitId was set to null, is this accepted?

**Review by yogesh-chauhan — APPROVED** [review]
> Reviewed 27 of 41 files at r8, 2 of 2 files at r9, 13 of 13 files at r10, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38690)* status: :shipit: complete! ✓

**Review by yogesh-chauhan — APPROVED** [review]

**Comment by yogesh-chauhan** [comment]
trigger codebasehash

**Review by yogesh-chauhan — APPROVED** [review]

**Comment by mohitg0795** [comment]
*[`136-git-sync-manager/src/main/java/io/harness/gitsync/common/impl/ScmFacilitatorServiceImpl.java` line 239 at r10](https://reviewable.io/reviews/harness/harness-core/38690#-NEVxNWk2km_lkyY3D8g:-NEWsMgl9alqHWtkvM7Y:bjr880s) ([raw file](https://github.com/harness/harness-core/blob/2ccabb31149525a02e6b21ca6775a662d1169dd8/136-git-sync-manager/src/main/java/io/harness/gitsync/common/impl/ScmFacilitatorServiceImpl.java#L239)):*
<details><summary><i>Previously, adivishy1 (Adithya Viswanathan) wrote…</i></summary><blockquote>

even earlier without your code changes as well, the comitId was set to null, is this accepted?
</blockquote></details>

ya basically, the method is getFileByBranch thus commitId is expected to be null in the request. The following method which we call supports both get file by commit and branch.

**Review by kapil-harness — DISMISSED** [review]
> Reviewed 27 of 41 files at r8, 2 of 2 files at r9, 13 of 13 files at r10, 1 of 1 files at r11, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38690)* status: 1 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained. 

___
*[`952-scm-java-client/src/main/java/io/harness/beans/response/scmBaseResponse.java` line 15 at r11](https://reviewable.io/reviews/harness/harness-core/38690#-NEWuvIxDnYCAAhj_Y-l:-NEWuvIyBg_G1QBM9F-u:bf2zq6b) ([raw file](https://github.com/harness/harness-core/blob/e0bec05c60a8f5ef8ced8647577b615594c03bce/952-scm-java-client/src/main/java/io/harness/beans/response/scmBaseResponse.java#L15)):*
> ```Java
> @SuperBuilder
> @Getter
> public abstract class scmBaseResponse {
> ```

class should start with CAPS

**Comment by mohitg0795** [comment]
*[`952-scm-java-client/src/main/java/io/harness/beans/response/scmBaseResponse.java` line 15 at r11](https://reviewable.io/reviews/harness/harness-core/38690#-NEWuvIxDnYCAAhj_Y-l:-NEWvtQF-uMPdV1-JaKI:bt2vr1b) ([raw file](https://github.com/harness/harness-core/blob/e0bec05c60a8f5ef8ced8647577b615594c03bce/952-scm-java-client/src/main/java/io/harness/beans/response/scmBaseResponse.java#L15)):*
<details><summary><i>Previously, kapil-harness (Kapil Choudhary) wrote…</i></summary><blockquote>

class should start with CAPS 
</blockquote></details>

Fixed.

**Review by kapil-harness — APPROVED** [review]
:lgtm:

> Reviewed 2 of 2 files at r12, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38690#-:-NEWwEop8NM293Uct47t:bnfp4nl)* status: 3 file(s) to review. 1 of 2 approvals obtained.

**Review by akhilpandey1 — APPROVED** [review]
> Reviewed 8 of 41 files at r8, 1 of 2 files at r9, 11 of 13 files at r10, 1 of 1 files at r11, 2 of 2 files at r12.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/38690)* status: :shipit: complete! ✓

**Pull request merged**